test/fixtures/google-gax-packaging-test-app/src/index.js

```diff
@@ -97,6 +102,57 @@ async function testEcho(client) {
   assert.deepStrictEqual(request.content, response.content);
 }
 
+async function testEchoError(client) {
+  const readFile = util.promisify(fs.readFile);
+
+  const fixtureName = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'test',
+    'fixtures',
+    'multipleErrors.json'
+  );
+  const protos_path = path.resolve(
+    __dirname,
+    '..',
+    '..',
+    '..',
+    'protos',
+    'google',
+    'rpc'
+  );
+
+  const data = await readFile(fixtureName, 'utf8');
+  const root = protobuf.loadSync(
+    path.join(protos_path, 'error_details.proto')
+  );
+  const objs = JSON.parse(data);
+  const MessageType = root.lookupType(objs[0].type);
+  const buffer = MessageType.encode(objs[0].value).finish();
+  const request = {
+    error: {
+      code: 3,
+      message: 'Test error',
+      details: [{
+        type_url: 'type.googleapis.com/' + objs[0].type,
+        value: buffer,
+      }],
+    },
+  };
+  const timer = setTimeout(() => {
+    throw new Error('End-to-end testEcho method fails with timeout');
+  }, 12000);
+  try {
```

Comment:
Instead of `try...catch`, we could use `assert.throw` to see if the method throw an error or not.